docs/proposals/release-notes.md

```diff
@@ -0,0 +1,194 @@
+interrupt-driven and often stressful process resulting in incomplete,
+inconsistent release notes often with errors and omissions.
+
+### The (general) Solution
+
+Like most things in the development/release pipeline, the earlier you do it,
+the easier it is for everyone and the better the outcome.  Gather your release
+notes earlier in the development cycle, at the time the features and fixes are
+added.
+
+This concept was first implemented at Google for the Borg team to
+capture release notes at submission time.  Since then the concept has
+been adopted widely and used successfully across Google.
+
+#### Then why not just list *every* change that was submitted, CHANGELOG-style?
+
+On larger projects like Kubernetes, showing every single change (PR) would mean
+hundreds of entries.  The goal is to highlight the major changes for a release.
+
+## Options
+
+1. Use of pre-commit and other local git hooks
+   * Experiments here using `prepare-commit-msg` and `commit-msg` git hook files
+     were promising but less than optimal due to the fact that they would
+     require input/confirmation with each commit and there may be multiple
+     commits in a push and eventual PR.
+1. Use of [github templates](https://github.com/blog/2111-issue-and-pull-request-templates)
+   * Templates provide a great way to pre-fill PR comments, but there are no
+     server-side hooks available to parse and/or easily check the contents of
+     those templates to ensure that checkboxes were checked or forms were filled
+     in.
+1. Use of labels enforced by mungers/bots
+   * We already make great use of mungers/bots to manage labels on PRs and it
+     fits very nicely in the existing workflow
+
+## Collection Design
+
+The munger/bot option fits most cleanly into the existing workflow.
+
+The design will include:
+
+1. New labels added to github: `release-note-NONE`, maybe others for new release note categories - see Layout section below
+1. A [new munger](https://github.com/kubernetes/kubernetes/issues/23409) that will:
+  * Initiate a `needs-release-note` label on all new PRs
+  * Disallow/Remove the `lgtm` label on all PRs labeled as `needs-release-note`
+  * Auto-remove `needs-release-note` when one of the release-note-\* labels is added
+  * Special case for cherry-picked/branch PRs, release-note-NONE is not allowed
+
+## Publishing Design
+
+### Location
+
+With v1.2.0, the release notes were moved from their previous [github releases](https://github.com/kubernetes/kubernetes/releases)
+location to [kubernetes/CHANGELOG.md](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md).  Going forward this seems like a good plan.
+Other projects do similar.
+
+The kuberntes.tar.gz download link is also displayed along with the release notes
+in [kubernetes/CHANGELOG.md](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md).
+
+Is there any reason to continue publishing anything to github releases if
```

Comment:
If the CHANGELOG contains the full release notes including links to the binary(s), what content do we want to have in the github releases?  Maybe just a link to CHANGELOG?